integrations/clickup/main.py

```diff
@@ -0,0 +1,117 @@
+from enum import StrEnum
+from loguru import logger
+from clickup.client import ClickUpClient
+
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+
+class ObjectKind(StrEnum):
+    """Defines the different kinds of objects in the system."""
+    TEAM = "team"
+    PROJECT = "project"
+    ISSUE = "issue"
+
+
+async def setup_application() -> None:
+    """
+    Sets up the application by creating necessary webhooks.
+
+    If no app host is provided, it logs a warning and skips the webhook creation.
+    """
+    logic_settings = ocean.integration_config
+    app_host = logic_settings.get("app_host")
+
+    if not app_host:
+        logger.warning(
+            "No app host provided, skipping webhook creation. "
+            "Without setting up the webhook, the integration will not export live changes from ClickUp."
+        )
+        return
+    #webhook  setup goes here
+
+@ocean.on_resync(ObjectKind.TEAM)
+async def on_resync_teams(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    """
+    Resyncs teams from ClickUp.
+
+    Yields:
+        List of team dictionaries.
+    """
+    client = ClickUpClient(
+        ocean.integration_config["clickup_host"],
+        ocean.integration_config["clickup_api_key"],
+    )
+
+    async for teams in client.get_paginated_teams():
+        logger.info(f"Received team batch with {len(teams)} teams")
+        yield teams
+
+
+@ocean.on_resync(ObjectKind.PROJECT)
+async def on_resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    """
+    Resyncs projects from ClickUp.
+
+    Yields:
+        List of project dictionaries.
+    """
+    client = ClickUpClient(
+        ocean.integration_config["clickup_host"],
+        ocean.integration_config["clickup_api_key"],
+    )
+
+    async for teams in client.get_paginated_teams():
+        for team in teams:
+            team_id = team["id"]
+            async for spaces in client.get_paginated_spaces(team_id):
+                logger.info(f"Received spaces batch with {len(spaces)} spaces for team {team_id}")
+                for space in spaces:
+                    space_id = space["id"]
```

Comment:
I would move this block to the client.py so that in this resync handler, I only call the get_paginated_projects() function